src/rust/engine/dep_inference/src/javascript/tests.rs

```diff
@@ -255,7 +255,7 @@ fn dynamic_scope() {
 #[test]
 fn adds_dir_to_file_imports() -> Result<(), Box<dyn std::error::Error>> {
     let result = get_dependencies(
-        &"import a from './file.js'",
+        "import a from './file.js'",
```

Comment:
Before I experiment, what about 1.75.0 makes this possible that wasn't before? I didn't see anything in the changelog that seemed directly relevant.

Reply:
Ditto the other code changes?

Reply:
This was doable before the upgrade, it is only now that they are flagged by `cargo clippy` as being able to be simplified.

Reply:
(All of the code changes are from clippy lints which are not triggered.)